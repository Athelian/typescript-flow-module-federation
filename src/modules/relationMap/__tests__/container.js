import RelationMapContainer from '../container';

describe('subTree', () => {
  test('is subTree', () => {
    const container = new RelationMapContainer();
    container.overrideState({
      targetedItem: {
        order: {
          order1: true,
        },
      },
    });
  });
});

describe('targetTree', () => {
  test('target Order => untarget Order', async () => {
    const container = new RelationMapContainer();
    const itemRelation = {
      order: {
        order1: true,
      },
      orderItem: {
        item1: true,
        item2: true,
      },
      batch: {
        batch1: true,
        batch2: true,
        batch3: true,
      },
    };
    const relation = { type: 'ORDER' };
    await container.targetTree(itemRelation, relation);
    expect(container.state).toMatchSnapshot();
    await container.resetTargetedItem(itemRelation, relation.type);
    expect(container.state).toMatchSnapshot();
    // console.log(JSON.stringify(container.state, null, 2))
  });

  test('target Order => untarget OrderItem => target Order', async () => {
    const container = new RelationMapContainer();
    const orderRelation = {
      order: {
        order1: true,
      },
      orderItem: {
        item1: true,
        item2: true,
      },
      batch: {
        batch1: true,
        batch2: true,
        batch3: true,
      },
    };
    const targetRelation = { type: 'ORDER' };

    const orderItemRelation = {
      order: {
        order1: true,
      },
      orderItem: {
        item1: true,
      },
      batch: {
        batch1: true,
      },
    };

    const untargetRelation = { type: 'ORDER_ITEM' };

    await container.reset();
    // 1. target order
    await container.targetTree(orderRelation, targetRelation);
    expect(container.state).toMatchSnapshot();
    // 2. untarget orderItem
    await container.resetTargetedItem(orderItemRelation, untargetRelation.type);
    expect(container.state).toMatchSnapshot();
    // 3. target order again
    await container.targetTree(orderRelation, targetRelation);
    expect(container.state).toMatchSnapshot();
  });

  test.only('target Order => untarget OrderItem => target OrderItem', async () => {
    const container = new RelationMapContainer();
    const orderRelation = {
      order: {
        order1: true,
      },
      orderItem: {
        item1: true,
        item2: true,
      },
      batch: {
        batch1: true,
        batch2: true,
        batch3: true,
      },
    };
    const targetRelation = { type: 'ORDER' };

    const orderItemRelation = {
      order: {
        order1: true,
      },
      orderItem: {
        item1: true,
      },
      batch: {
        batch1: true,
      },
    };

    const untargetRelation = { type: 'ORDER_ITEM' };

    await container.reset();
    // 1. target order
    await container.targetTree(orderRelation, targetRelation);
    // expect(container.state).toMatchSnapshot()
    // 2. untarget orderItem
    await container.resetTargetedItem(orderItemRelation, untargetRelation.type);
    // expect(container.state).toMatchSnapshot()
    // 3. target order again
    await container.targetTree(orderItemRelation, untargetRelation.type);
    console.log(JSON.stringify(container.state, null, 2));
    // expect(container.state).toMatchSnapshot()
  });
});
