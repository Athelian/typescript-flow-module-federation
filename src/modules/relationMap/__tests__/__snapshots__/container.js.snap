// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`targetTree target Order => untarget Order 1`] = `
Object {
  "focusMode": "TARGET_TREE",
  "focusedId": "",
  "focusedItem": Object {
    "batch": Object {},
    "order": Object {},
    "orderItem": Object {},
    "shipment": Object {},
  },
  "lines": Array [
    Object {
      "batch1": Object {
        "id": "batch1",
        "line": true,
        "related": true,
      },
      "batch2": Object {
        "id": "batch2",
        "line": true,
        "related": true,
      },
      "batch3": Object {
        "id": "batch3",
        "line": true,
        "related": true,
      },
      "item1": Object {
        "id": "item1",
        "line": true,
        "related": true,
      },
      "item2": Object {
        "id": "item2",
        "line": true,
        "related": true,
      },
      "order1": Object {
        "id": "order1",
        "line": true,
        "related": true,
      },
    },
  ],
  "targetedItem": Object {
    "batch": Object {
      "batch1": true,
      "batch2": true,
      "batch3": true,
    },
    "order": Object {
      "order1": true,
    },
    "orderItem": Object {
      "item1": true,
      "item2": true,
    },
  },
  "trees": Array [
    Object {
      "batch1": true,
      "batch2": true,
      "batch3": true,
      "item1": true,
      "item2": true,
      "order1": true,
    },
  ],
}
`;

exports[`targetTree target Order => untarget Order 2`] = `
Object {
  "focusMode": "TARGET",
  "focusedId": "",
  "focusedItem": Object {
    "batch": Object {},
    "order": Object {},
    "orderItem": Object {},
    "shipment": Object {},
  },
  "lines": Array [],
  "targetedItem": Object {
    "batch": Object {},
    "order": Object {},
    "orderItem": Object {},
  },
  "trees": Array [
    Object {},
  ],
}
`;

exports[`targetTree target Order => untarget OrderItem => target Order 1`] = `
Object {
  "focusMode": "TARGET_TREE",
  "focusedId": "",
  "focusedItem": Object {
    "batch": Object {},
    "order": Object {},
    "orderItem": Object {},
    "shipment": Object {},
  },
  "lines": Array [
    Object {
      "batch1": Object {
        "id": "batch1",
        "line": true,
        "related": true,
      },
      "batch2": Object {
        "id": "batch2",
        "line": true,
        "related": true,
      },
      "batch3": Object {
        "id": "batch3",
        "line": true,
        "related": true,
      },
      "item1": Object {
        "id": "item1",
        "line": true,
        "related": true,
      },
      "item2": Object {
        "id": "item2",
        "line": true,
        "related": true,
      },
      "order1": Object {
        "id": "order1",
        "line": true,
        "related": true,
      },
    },
  ],
  "targetedItem": Object {
    "batch": Object {
      "batch1": true,
      "batch2": true,
      "batch3": true,
    },
    "order": Object {
      "order1": true,
    },
    "orderItem": Object {
      "item1": true,
      "item2": true,
    },
  },
  "trees": Array [
    Object {
      "batch1": true,
      "batch2": true,
      "batch3": true,
      "item1": true,
      "item2": true,
      "order1": true,
    },
  ],
}
`;

exports[`targetTree target Order => untarget OrderItem => target Order 2`] = `
Object {
  "focusMode": "TARGET",
  "focusedId": "",
  "focusedItem": Object {
    "batch": Object {},
    "order": Object {},
    "orderItem": Object {},
    "shipment": Object {},
  },
  "lines": Array [
    Object {
      "batch1": Object {
        "id": "batch1",
        "line": false,
        "related": false,
      },
      "batch2": Object {
        "id": "batch2",
        "line": true,
        "related": true,
      },
      "batch3": Object {
        "id": "batch3",
        "line": true,
        "related": true,
      },
      "item1": Object {
        "id": "item1",
        "line": true,
        "related": false,
      },
      "item2": Object {
        "id": "item2",
        "line": true,
        "related": true,
      },
      "order1": Object {
        "id": "order1",
        "line": true,
        "related": false,
      },
    },
  ],
  "targetedItem": Object {
    "batch": Object {
      "batch2": true,
      "batch3": true,
    },
    "order": Object {
      "order1": true,
    },
    "orderItem": Object {
      "item2": true,
    },
  },
  "trees": Array [
    Object {
      "batch2": true,
      "batch3": true,
      "item2": true,
    },
  ],
}
`;

exports[`targetTree target Order => untarget OrderItem => target Order 3`] = `
Object {
  "focusMode": "TARGET_TREE",
  "focusedId": "",
  "focusedItem": Object {
    "batch": Object {},
    "order": Object {},
    "orderItem": Object {},
    "shipment": Object {},
  },
  "lines": Array [
    Object {
      "batch1": Object {
        "id": "batch1",
        "line": false,
        "related": false,
      },
      "batch2": Object {
        "id": "batch2",
        "line": true,
        "related": true,
      },
      "batch3": Object {
        "id": "batch3",
        "line": true,
        "related": true,
      },
      "item1": Object {
        "id": "item1",
        "line": true,
        "related": false,
      },
      "item2": Object {
        "id": "item2",
        "line": true,
        "related": true,
      },
      "order1": Object {
        "id": "order1",
        "line": true,
        "related": false,
      },
    },
    Object {
      "batch1": Object {
        "id": "batch1",
        "line": true,
        "related": true,
      },
      "batch2": Object {
        "id": "batch2",
        "line": true,
        "related": true,
      },
      "batch3": Object {
        "id": "batch3",
        "line": true,
        "related": true,
      },
      "item1": Object {
        "id": "item1",
        "line": true,
        "related": true,
      },
      "item2": Object {
        "id": "item2",
        "line": true,
        "related": true,
      },
      "order1": Object {
        "id": "order1",
        "line": true,
        "related": true,
      },
    },
  ],
  "targetedItem": Object {
    "batch": Object {
      "batch1": true,
      "batch2": true,
      "batch3": true,
    },
    "order": Object {
      "order1": true,
    },
    "orderItem": Object {
      "item1": true,
      "item2": true,
    },
  },
  "trees": Array [
    Object {
      "batch1": true,
      "batch2": true,
      "batch3": true,
      "item1": true,
      "item2": true,
      "order1": true,
    },
  ],
}
`;

exports[`targetTree target Order => untarget OrderItem => target OrderItem 1`] = `
Object {
  "focusMode": "TARGET_TREE",
  "focusedId": "",
  "focusedItem": Object {
    "batch": Object {},
    "order": Object {},
    "orderItem": Object {},
    "shipment": Object {},
  },
  "lines": Array [
    Object {
      "batch1": Object {
        "id": "batch1",
        "line": true,
        "related": true,
        "type": "BATCH",
      },
      "batch2": Object {
        "id": "batch2",
        "line": true,
        "related": true,
        "type": "BATCH",
      },
      "batch3": Object {
        "id": "batch3",
        "line": true,
        "related": true,
        "type": "BATCH",
      },
      "item1": Object {
        "id": "item1",
        "line": true,
        "related": true,
        "type": "ORDER_ITEM",
      },
      "item2": Object {
        "id": "item2",
        "line": true,
        "related": true,
        "type": "ORDER_ITEM",
      },
      "order1": Object {
        "id": "order1",
        "line": true,
        "related": true,
        "type": "ORDER",
      },
    },
  ],
  "targetedItem": Object {
    "batch": Object {
      "batch1": true,
      "batch2": true,
      "batch3": true,
    },
    "order": Object {
      "order1": true,
    },
    "orderItem": Object {
      "item1": true,
      "item2": true,
    },
  },
  "trees": Array [
    Object {
      "batch1": true,
      "batch2": true,
      "batch3": true,
      "item1": true,
      "item2": true,
      "order1": true,
    },
  ],
}
`;

exports[`targetTree target Order => untarget OrderItem => target OrderItem 2`] = `
Object {
  "focusMode": "TARGET",
  "focusedId": "",
  "focusedItem": Object {
    "batch": Object {},
    "order": Object {},
    "orderItem": Object {},
    "shipment": Object {},
  },
  "lines": Array [
    Object {
      "batch1": Object {
        "id": "batch1",
        "line": true,
        "related": true,
        "type": "BATCH",
      },
      "batch2": Object {
        "id": "batch2",
        "line": true,
        "related": true,
        "type": "BATCH",
      },
      "batch3": Object {
        "id": "batch3",
        "line": true,
        "related": true,
        "type": "BATCH",
      },
      "item1": Object {
        "id": "item1",
        "line": true,
        "related": true,
        "type": "ORDER_ITEM",
      },
      "item2": Object {
        "id": "item2",
        "line": true,
        "related": true,
        "type": "ORDER_ITEM",
      },
      "order1": Object {
        "id": "order1",
        "line": true,
        "related": true,
        "type": "ORDER",
      },
    },
  ],
  "targetedItem": Object {
    "batch": Object {
      "batch2": true,
      "batch3": true,
    },
    "order": Object {},
    "orderItem": Object {
      "item2": true,
    },
  },
  "trees": Array [
    Object {
      "batch2": true,
      "batch3": true,
      "item2": true,
    },
  ],
}
`;

exports[`targetTree target Order => untarget OrderItem => target OrderItem 3`] = `
Object {
  "focusMode": "TARGET_TREE",
  "focusedId": "",
  "focusedItem": Object {
    "batch": Object {},
    "order": Object {},
    "orderItem": Object {},
    "shipment": Object {},
  },
  "lines": Array [
    Object {
      "batch1": Object {
        "id": "batch1",
        "line": true,
        "related": true,
        "type": "BATCH",
      },
      "item1": Object {
        "id": "item1",
        "line": true,
        "related": true,
        "type": "ORDER_ITEM",
      },
      "order1": Object {
        "id": "order1",
        "line": true,
        "related": true,
        "type": "ORDER",
      },
    },
  ],
  "targetedItem": Object {
    "batch": Object {
      "batch1": true,
      "batch2": true,
      "batch3": true,
    },
    "order": Object {
      "order1": true,
    },
    "orderItem": Object {
      "item1": true,
      "item2": true,
    },
  },
  "trees": Array [
    Object {
      "batch2": true,
      "batch3": true,
      "item2": true,
    },
    Object {
      "batch1": true,
      "item1": true,
      "order1": true,
    },
  ],
}
`;
